runtime/oti/jvmtiInternal.h

```diff
@@ -482,6 +482,22 @@ typedef struct jvmtiGcp_translation {
 		} \
 	} while(0)
 
+#if JAVA_SPEC_VERSION >= 19
+#define ENSURE_JTHREAD_NOT_VIRTUAL(vmThread, jthrd) \
+	do { \
+		if (isSameOrSuperClassOf(J9VMJAVALANGBASEVIRTUALTHREAD_OR_NULL((vmThread->javaVM)), J9OBJECT_CLAZZ((vmThread), *((j9object_t *) (jthrd))))) { \
```

Comment:
can you add a common helper for this in j9.h

something like `J9VM_IS_CURRENT_THREAD_VIRTUAL(vmthread)` and `J9_IS_JAVALANGTHREAD_VIRTUAL`

Reply:
Do you mean one or two helpers?
Something like
```
#define J9VM_IS_CURRENT_THREAD_VIRTUAL(vmThread, threadObj) isSameOrSuperClassOf(J9VMJAVALANGBASEVIRTUALTHREAD_OR_NULL((vmThread->javaVM)), J9OBJECT_CLAZZ((vmThread), threadObj))
```
?

Reply:
`IS_VIRTUAL_THREAD(...)`. See below example: https://github.com/eclipse-openj9/openj9/blob/43258c1b2adbc3bff44340b0c126d723c7d530ba/runtime/oti/j9.h#L132-L135